lib/api/apiUtils/authorization/prepareRequestContexts.js

```diff
@@ -118,6 +118,12 @@ function prepareRequestContexts(apiMethod, request, sourceBucket,
               generateRequestContext('objectPutACL');
             requestContexts.push(putAclRequestContext);
         }
+
+        if (request.headers['x-amz-object-lock-mode']) {
+            const putObjectLockRequestConext =
+              generateRequestContext('objectPutRetention');
+            requestContexts.push(putObjectLockRequestConext);
```

Comment:
Small typo:

```suggestion
            const putObjectLockRequestContext =
              generateRequestContext('objectPutRetention');
            requestContexts.push(putObjectLockRequestContext);
```

Reply:
oof, good catch, maybe the queue is stuck and I can merge this as well...